### PR TITLE
add new resource auth0_branding

### PR DESCRIPTION
**Pull request by gagalago**
continuation of https://github.com/alexkappa/terraform-provider-auth0/pull/199 + universal login template from https://github.com/go-auth0/auth0/pull/161

### Proposed Changes

* add branding resource
* add branding universal login template

related to https://github.com/alexkappa/terraform-provider-auth0/issues/124
fix https://github.com/alexkappa/terraform-provider-auth0/issues/274

#### Acceptance Test Output

```
$ make testacc https://github.com/go-auth0/auth0/pull/161

...
```

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,8 @@ module github.com/alexkappa/terraform-provider-auth0
 
 go 1.13
 
+replace gopkg.in/auth0.v5 => github.com/sortlist/auth0-1 v0.0.0-20201125100601-156820b66f1b
```
Comment (relu) [inline go.mod:5]: TODO: Remove this once the new version of the sdk is released.

File: go.sum [hunk 2]
```diff
@@ -322,6 +322,14 @@ github.com/sergi/go-diff v1.0.0 h1:Kpca3qRNrduNnOQeazBd0ysaKrUJiIuISHxogkT9RPQ=
 github.com/sergi/go-diff v1.0.0/go.mod h1:0CfEIISq7TuYL3j771MWULgwwjU+GofnZX9QAmXWZgo=
 github.com/sergi/go-diff v1.1.0 h1:we8PVUC3FE2uYfodKH/nBHMSetSfHDR6scGdBi+erh0=
 github.com/sergi/go-diff v1.1.0/go.mod h1:STckp+ISIX8hZLjrqAeVduY0gWCT9IjLuqbuNXdaHfM=
+github.com/sortlist/auth0-1 v0.0.0-20201112090129-3530f2dc4609 h1:nrj/VNFAt7M19A+0rA9HUauLzAn3CL1UtW4sRVaopQE=
```
Comment (relu) [inline go.sum:325]: TODO: Please run `go mod tidy` and commit the new sum file.

**Comment by adomokos** [comment]
Hello! We would love to use this functionality as we are using their mgmt API to set Universal Login Page manually. Is there anything blocking this?

**Comment by gagalago** [comment]
> Hello! We would love to use this functionality as we are using their mgmt API to set Universal Login Page manually. Is there anything blocking this?

I cannot make tests pass, go error is a bit too cryptic to me (I am still a go beginner):
```
make testacc TESTS=TestAccBranding           
=== RUN   TestAccBranding
    testing.go:705: Step 0 error: Check failed: Check 6/6 error: auth0_branding.my_branding: Attribute 'universal_login.body' not found
--- FAIL: TestAccBranding (2.32s)
```

**Comment by adomokos** [comment]
Woohoo! @relu - any thoughts about this and maybe getting it out?

**Comment by relu** [comment]
@adomokos I'm just here for "moral support", @alexkappa would be the right person to put the Official Stamp:tm:  on it :laughing:. Cheers!

**Comment by adomokos** [comment]
@alexkappa, can you take a look when you have a moment? We'd love to set up branding with tf.

**Review by relu — APPROVED** [review]
If it matters at all, LGTM! :heavy_check_mark:

**Comment by adomokos** [comment]
Ouch.

**Comment by rabidscorpio** [comment]
@alexkappa Can you look at this? This would be a nice addition.

**Comment by adomokos** [comment]
@alexkappa - pinging you here again. This was set to stale and auto-closed. Can you please review it and (maybe) merge it/release it? We could use this functionality right away. Thanks!

**Comment by toshitanian** [comment]
👍

**Comment by jbauth0** [comment]
I'll defer to @alexkappa but I feel that the `/api/v2/branding` settings/endpoint should be separate from the `/api/v2/branding/templates/universal-login` settings/endpoint.

That being said I think the code looks good.

**Comment by rabidscorpio** [comment]
@yvovandoorn Can you take a look at this since it was only closed for being stale?

**Comment by yvovandoorn** [comment]
Hey @rabidscorpio & @gagalago. It appears there are now two PRs that tackle branding. This one and #357 (by @toshi1127 ).

Comparing PRs, it looks like I can merge #357 as it doesn't require `universal_login` which is an optional body but it marked as required in this PR. Any objection if this remains closed and I proceed to merge in #357?

**Comment by gagalago** [comment]
no problem for me. what matter, is that we can now use the branding endpoint :tada:

**Pull request closed without merging**